src/Microsoft.AspNetCore.SignalR.Client.TS/Transports.ts

```diff
@@ -160,7 +173,21 @@ export class LongPollingTransport implements ITransport {
         pollXhr.onload = () => {
             if (pollXhr.status == 200) {
                 if (this.onDataReceived) {
-                    this.onDataReceived(pollXhr.response);
+                    // Parse the messages
+                    let [error, messages] = Formatters.parseMessages(pollXhr.response);
+                    if (error) {
+                        if (this.onError) {
+                            // TODO: Is this how we should emit this?
```

Comment:
This is OK for now. There is a [_unify errors_](https://github.com/aspnet/SignalR/issues/3) item on my list which is about making sure that we actually use Error wherever possible/practical.